dask/array/routines.py

```diff
@@ -929,6 +932,304 @@ def histogram(a, bins=None, range=None, normed=False, weights=None, density=None
+        explicitly in ``bins``. If defined, this argument is required
+        to have an entry for each dimension. Unlike
+        :func:`numpy.histogramdd`, if `bins` does not define bin
+        edges, this argument is required (this function will not
+        automatically use the min and max of of the value in a given
+        dimension because the input data may be lazy in dask).
+    normed : bool, optional
+        An alias for the density argument that behaves identically. To
+        avoid confusion with the broken argument to `histogram`,
+        `density` should be preferred.
+    weights : dask Array, optional
+        An array of values weighing each sample in the input data. The
+        chunks of the the weights must be identical to the chunking
+        along the 0th axis of the data sample.
+    density : bool, optional
+        If ``False`` (default), the returned array represents the
+        number of samples in each bin. If ``True``, the returned array
+        represents the probability density function at each bin.
+
+    See Also
+    --------
+    histogram
+
+    Examples
+    --------
+    Computing the histogram in 5 blocks using different bin edges
+    along each dimension:
+
+    >>> import dask.array as da
+    >>> x = da.random.uniform(0, 1, size=(1000, 3), chunks=(200, 3))
+    >>> edges = [
+    ...     np.linspace(0, 1, 5), # 4 bins in 1st dim
+    ...     np.linspace(0, 1, 6), # 5 in the 2nd
+    ...     np.linspace(0, 1, 4), # 3 in the 3rd
+    ... ]
+    >>> h, edges = da.histogramdd(x, bins=edges)
+    >>> result = h.compute()
+    >>> result.shape
+    (4, 5, 3)
+
+    Defining the bins by total number and their ranges, along with
+    using weights:
+
+    >>> bins = (4, 5, 3)
+    >>> ranges = ((0, 1),) * 3  # expands to ((0, 1), (0, 1), (0, 1))
+    >>> w = da.random.uniform(0, 1, size=(1000,), chunks=x.chunksize[0])
+    >>> h, edges = da.histogramdd(x, bins=bins, range=ranges, weights=w)
+    >>> np.isclose(h.sum().compute(), w.sum().compute())
+    True
+
+    """
+
+    # logic used in numpy.histogramdd to handle normed/density.
+    if normed is None:
+        if density is None:
+            density = False
+    elif density is None:
+        # an explicit normed argument was passed, alias it to the new name
+        density = normed
+    else:
+        raise TypeError("Cannot specify both 'normed' and 'density'")
+
+    # generate token and name for task
+    token = tokenize(sample, bins, range, weights, density)
+    name = f"histogramdd-sum-{token}"
+
+    # N == total number of samples
+    # D == total number of dimensions
+    try:
+        # Recommended input ND-array
+        N, D = sample.shape
+    except (AttributeError, ValueError):
+        # If we have a sequence of 1D arrays
+        sample = atleast_2d(sample).T
+        N, D = sample.shape
+        # rechunk if necessary
+        if sample.chunksize[1] != D:
+            sample = sample.rechunk((sample.chunksize[1], D))
```

Comment:
Not to my knowledge.

Taking a 3D example: if `sample` is a list of 1D vectors `x`, `y`, and `z`, we have to stitch the transposed-array-of-vectors together such that each `x[i]`, `y[i]`, and `z[i]` live on the same chunk so they can be passed to `np.histogramdd` together. If `sample.chunksize[1] != 3` (in this example with x, y, z), we have to rechunk if we want to use this function.

Side note: this should definitely be `rechunk((sample.chunksize[0], D))`! Fixing this now, and adding a test that follows this rechunking code-path

Reply:
No worries. This seems like a reasonable approach. The main reason for the question is that rechunking in Dask can incur some overhead. It may involve performing some communication and hanging onto more memory in fewer workers. So it is something worth avoiding when possible

That all being said, when there is not a clear alternative, rechunking can be reasonable. It's hard to say whether this will be much of a bottleneck in reality without seeing some people using this and providing feedback. If it does present an issue later, we can always revisit and rethink our approach here. For now leaving this as-is seems fine 🙂